core/arch/arm/mm/core_mmu_v7.c

```diff
@@ -600,6 +600,18 @@ static void map_memarea(struct tee_mmap_region *mm, uint32_t *ttb)
 
 	TEE_ASSERT(mm && ttb);
 
+	/*
+	 * If mm->va is small than 32M, means mm->va conflicts with
+	 * user ta address. It may cause mm->va can not be used or corrupt
+	 * system, since later when user ta loaded, mm->va is destroyed
+	 * by user ta. Need to reset mm->va, but not directly mm->va = mm->pa.
+	 * Some SoCs's device address may be lower than 32MB.
+	 *
+	 * TODO: fix when mm->va < 32M.
+	 */
+	if (mm->va < (TEE_MMU_UL1_NUM_ENTRIES * SECTION_SIZE))
+		EMSG("va 0x%" PRIxVA " conflicts with user ta address!",
+		     mm->va);
```

Comment:
I guess we should panic() here since it will only lead to troubles if continuing.